### DISCUSSION TITLE
Is this framework fully compatible with standard ONNX releases?

### DISCUSSION THREAD

**Discussion by albertoandreottiATgmail** [fastmachinelearning/qonnx#63] [General]
Hello,

I think the answer to this question is probably 'no'. But, I opened it to discuss, so this project provides new onnx 'nodes' that work with quantization, but we need a custom engine to run it?
Any plan to merge these new operations into the main ONNX project?
Thanks for any help, and in case I'm misreading the goals here, please educate me accordingly.

Alberto.

**Comment by maltanar** [comment]
Hi Alberto, the quantization ops proposed by QONNX such as `Quant` and `BipolarQuant` are not part of the ONNX standard. Whether we want to make a proposal to get them adopted into the standard is under discussion in the community, mostly because getting something into the standard (and maintaining it afterwards) is a lot of effort that we don't necessarily have the resources for, and because the quantization/datatype requirements in the ML world is still in flux. 

So as you observed, the custom engine in this repo is currently required to e.g. run inference on graphs with `Quant` nodes, or to transform them in some cases to standards-compatible representations like Q(C)DQ.
